src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/LengthConverter.cs

```diff
@@ -191,22 +176,21 @@ static internal void FormatLengthAsString(double value, ref DefaultInterpolatedS
         //   [value] is a double
         //   [unit] is a string specifying the unit, like 'in' or 'px', or nothing (means pixels)
         // NOTE - This code is called from FontSizeConverter, so changes will affect both.
-        static internal double FromString(string s, CultureInfo cultureInfo)
+        internal static double FromString(ReadOnlySpan<char> value, CultureInfo cultureInfo)
         {
-            ReadOnlySpan<char> valueSpan = s.AsSpan().Trim();
+            ReadOnlySpan<char> valueSpan = value.Trim();
             double unitFactor = 1.0;
 
-            //Auto is represented and Double.NaN
-            //properties that do not want Auto and NaN to be in their ligit values,
-            //should disallow NaN in validation callbacks (same goes for negative values)
-            if (valueSpan.Equals("auto", StringComparison.OrdinalIgnoreCase))
-                return Double.NaN;
-
-            PixelUnit pixelUnit;
-            if (PixelUnit.TryParsePixel(valueSpan, out pixelUnit)
-                || PixelUnit.TryParsePixelPerInch(valueSpan, out pixelUnit)
```

Comment:
Is the `||` or `&&` at the start of the line instead of at the end of the line wrong according to the coding style ? Both seems to be used a lot and if one isn't better than the other I don't see a reason to change it.